libheif/api/libheif/heif_emscripten.h

```diff
@@ -99,6 +100,48 @@ static emscripten::val heif_js_context_get_list_of_top_level_image_IDs(
   return result;
 }
 
+
+static emscripten::val heif_js_context_get_list_of_item_IDs(
+    struct heif_context* context)
+{
+  emscripten::val result = emscripten::val::array();
+  if (!context) {
+    return result;
+  }
+
+  int count = heif_context_get_number_of_items(context);
+  if (count <= 0) {
+    return result;
+  }
+
+  heif_item_id* ids = (heif_item_id*) malloc(count * sizeof(heif_item_id));
```

Comment:
Just an idea: you could consider using `alloca()` instead of `malloc()`. That is ideal for these small chunks of temporary memory. I'm not sure whether it's available within emscripten, though.